builtin/logical/aws/client.go

```diff
@@ -29,6 +30,16 @@ func getRootConfig(s logical.Storage) (*aws.Config, error) {
 		credsConfig.AccessKey = config.AccessKey
 		credsConfig.SecretKey = config.SecretKey
 		credsConfig.Region = config.Region
+		credsConfig.IAMEndpoint = config.IAMEndpoint
+		credsConfig.STSEndpoint = config.STSEndpoint
+
+		if clientType == "iam" && config.IAMEndpoint != "" {
+			endpoint = *aws.String(config.IAMEndpoint)
+		}
+
+		if clientType == "sts" && config.STSEndpoint != "" {
+			endpoint = *aws.String(config.STSEndpoint)
+		}
```

Comment:
It would be more go-idiomatic to do something like:

```go
switch {
case clientType == "iam" && config.IAMEndpoint != "":
	endpoint = *aws.String(config.IAMEndpoint)
case clientType == "sts" && config.STSEndpoint != "":
	endpoint = *aws.String(config.STSEndpoint)
}
```